frontend/src/components/GoogleMaps/LocationSearch.tsx

```diff
@@ -1,39 +1,53 @@
 import { GooglePlacesAutocomplete } from "react-native-google-places-autocomplete";
 import React, { useRef, useState } from "react";
 import { View, Text, FlatList, TouchableOpacity } from "react-native";
-const GOOGLE_MAP_API_KEY = process.env.EXPO_PUBLIC_GOOGLE_MAP_API_KEY || "undefined";
-import { DataItem } from '@/types'; import { AntDesign } from "@expo/vector-icons";
-;
-
-export const LocationSearch = (
-  { onLocationSelected, value }:
-    {
-      onLocationSelected: (location: { address: string, citystate: string; }) => void;
-      value: string;
-    }) => {
-
+const GOOGLE_MAP_API_KEY =
+  process.env.EXPO_PUBLIC_GOOGLE_MAP_API_KEY || "undefined";
+import { DataItem } from "@/types";
+import { AntDesign } from "@expo/vector-icons";
+export const LocationSearch = ({
+  onLocationSelected,
+  onChange,
+  value,
+}: {
+  onLocationSelected: (location: {
+    address: string;
+    citystate: string;
+  }) => void;
+  onChange: (text: string) => void;
+  value: string;
+}) => {
   const data = [
     {
       id: "googlePlacesInput",
-      component:
+      component: (
         <GooglePlacesAutocomplete
           textInputProps={{
-            value: value === 'undefined' ? null : value,
+            value: value === "undefined" ? null : value,
+            onChangeText: onChange,
           }}
-          placeholder='Search a location'
+          placeholder="Search a location"
           onPress={(data, details = null) => {
-            console.log("location", data);
-            console.log("street", data.structured_formatting.main_text);
-            console.log("city-level location", data.structured_formatting.secondary_text);
+            // console.log("location", data);
+            // console.log("street", data.structured_formatting.main_text);
+            // console.log(
+            //   "city-level location",
+            //   data.structured_formatting.secondary_text,
+            // );
             onLocationSelected({
               address: data.structured_formatting.main_text,
               citystate: data.structured_formatting.secondary_text,
             });
+            onChange(
+              data.structured_formatting.main_text +
+                " " +
+                data.structured_formatting.secondary_text,
```

Comment:
Is there any reason why we need `onChange` handler? Seems like you can get the same information from `onLocationSelected`.

Reply:
onChange here is to set the user input (value) to match the results (LocationSelected), which also allows user to edit the input later.